File: Demos/优化/RainForestSwift/RainForestSwift/Factories.m

#import "Factories.h"

@implementation NSAttributedString (Factories)
+ (NSAttributedString *)attributedStringForTitleText:(NSString *)text
{
    NSDictionary *titleAttributes = @{NSFontAttributeName: [UIFont fontWithName:@"AvenirNext-Heavy" size:30], NSForegroundColorAttributeName: [UIColor whiteColor], NSShadowAttributeName: [NSShadow titleTextShadow], NSParagraphStyleAttributeName: [NSParagraphStyle justifiedParagraphStyle]};
    
    return [[NSAttributedString alloc] initWithString:text attributes:titleAttributes];
}

+ (NSAttributedString *)attributedStringForDescription:(NSString *)text
{
    NSDictionary *descriptionAttributes = @{NSFontAttributeName: [UIFont fontWithName:@"AvenirNext-Medium" size:16],
                                      NSForegroundColorAttributeName: [UIColor whiteColor],
                                      NSBackgroundColorAttributeName: [UIColor clearColor],
                                      NSShadowAttributeName: [NSShadow descriptionTextShadow], NSParagraphStyleAttributeName: [NSParagraphStyle justifiedParagraphStyle]};
    
    return [[NSAttributedString alloc] initWithString:text attributes:descriptionAttributes];
}
@end

@implementation NSParagraphStyle (Factories)
+ (NSParagraphStyle *)justifiedParagraphStyle
{
    NSMutableParagraphStyle *paragraphStlye = [[NSMutableParagraphStyle alloc] init];
    paragraphStlye.alignment = NSTextAlignmentJustified;
    return paragraphStlye;
}

@end

@implementation NSShadow (Factories)
+ (NSShadow *)titleTextShadow
{
    NSShadow *shadow = [[NSShadow alloc] init];
    shadow.shadowColor = [UIColor colorWithHue:0 saturation:0 brightness:0 alpha:0.3];
    shadow.shadowOffset = CGSizeMake(0, 2);
    shadow.shadowBlurRadius = 3.0;
    
    return shadow;
}

+ (NSShadow *)descriptionTextShadow
{
    NSShadow *shadow = [[NSShadow alloc] init];
    shadow.shadowColor = [UIColor colorWithWhite:0.0 alpha:0.3];
    shadow.shadowOffset = CGSizeMake(0, 1);
    shadow.shadowBlurRadius = 3.0;
    
    return shadow;
}
@end

File: Demos/优化/RainForestSwift/RainForestSwift/Factories.swift
//
//  Factories.swift
//  AsyncDisplay
//
//  Created by 伯驹 黄 on 2017/4/20.
//  Copyright © 2017年 伯驹 黄. All rights reserved.
//

extension NSParagraphStyle {
    static var justifiedParagraphStyle: NSParagraphStyle {
        let paragraphStlye = NSMutableParagraphStyle()
        paragraphStlye.alignment = .justified
        return paragraphStlye
    }
}

extension NSShadow {
    static var titleTextShadow: NSShadow {
        let shadow = NSShadow()
        shadow.shadowColor = UIColor(hue: 0, saturation: 0, brightness: 0, alpha: 0.3)
        shadow.shadowOffset = CGSize(width: 0, height: 2)
        shadow.shadowBlurRadius = 3.0
        return shadow
    }

    static var descriptionTextShadow: NSShadow {
        let shadow = NSShadow()
        shadow.shadowColor = UIColor(white: 0, alpha: 0.3)
        shadow.shadowOffset = CGSize(width: 0, height: 1)
        shadow.shadowBlurRadius = 3.0
        
        return shadow
    }
}

extension NSAttributedString {
    convenience init(forTitleText text: String) {

        let titleAttributes: [String: Any] = [
            NSFontAttributeName: UIFont(name: "AvenirNext-Heavy", size: 30)!,
            NSForegroundColorAttributeName: UIColor.white,
            NSShadowAttributeName: NSShadow.titleTextShadow,
            NSParagraphStyleAttributeName: NSParagraphStyle.justifiedParagraphStyle
        ]
        self.init(string: text, attributes: titleAttributes)

    }

    convenience init(forDescription text: String) {
        let descriptionAttributes: [String: Any] = [
            NSFontAttributeName: UIFont(name: "AvenirNext-Medium", size: 16)!,
            NSForegroundColorAttributeName: UIColor.white,
            NSBackgroundColorAttributeName: UIColor.clear,
            NSShadowAttributeName: NSShadow.descriptionTextShadow, NSParagraphStyleAttributeName: NSParagraphStyle.justifiedParagraphStyle
        ]

        self.init(string: text, attributes: descriptionAttributes)
    }
}
